frontend/src/metabase/nav/components/SearchBar.unit.spec.tsx

```diff
@@ -0,0 +1,182 @@
+import { Route } from "react-router";
+import userEvent from "@testing-library/user-event";
+import { waitFor, renderWithProviders, screen, within } from "__support__/ui";
+import {
+  setupRecentViewsEndpoints,
+  setupSearchEndpoints,
+} from "__support__/server-mocks";
+import {
+  createMockCollectionItem,
+  createMockModelObject,
+  createMockRecentItem,
+} from "metabase-types/api/mocks";
+import {
+  createMockSettingsState,
+  createMockState,
+} from "metabase-types/store/mocks";
+import { CollectionItem, RecentItem } from "metabase-types/api";
+import SearchBar from "metabase/nav/components/SearchBar";
+
+const TEST_SEARCH_RESULTS: CollectionItem[] = [
+  "Card ABC",
+  "Card BCD",
+  "Card DEF",
+  "Card EFG",
+].map((name, index) =>
+  createMockCollectionItem({
+    name,
+    id: index + 1,
+    getUrl: () => "/",
+  }),
+);
+
+const TEST_RECENT_VIEWS_RESULTS: RecentItem[] = [
+  "Recents ABC",
+  "Recents BCD",
+  "Recents CDE",
+  "Recents DEF",
+].map((name, index) =>
+  createMockRecentItem({
+    model_object: createMockModelObject({ name }),
+    model_id: index,
+  }),
+);
+
+const setup = ({
+  initialRoute = "/",
+  searchResultItems = TEST_SEARCH_RESULTS,
+  recentViewsItems = TEST_RECENT_VIEWS_RESULTS,
+} = {}) => {
+  const state = createMockState({
+    settings: createMockSettingsState({
+      "search-typeahead-enabled": true,
+    }),
+  });
+
+  setupSearchEndpoints(searchResultItems);
+  setupRecentViewsEndpoints(recentViewsItems);
+
+  const { history } = renderWithProviders(
+    <Route path="*" component={SearchBar} />,
+    {
+      withRouter: true,
+      initialRoute,
+      storeInitialState: state,
+    },
+  );
+
+  return { history };
+};
+
+const getSearchBar = () => {
+  return screen.getByPlaceholderText("Search…");
+};
+
+describe("SearchBar", () => {
+  describe("typing a search query", () => {
+    it("should change URL when user types a query and hits `Enter`", async () => {
+      const { history } = setup();
+
+      userEvent.type(getSearchBar(), "BC{enter}");
+
+      const location = history?.getCurrentLocation();
+      expect(location?.pathname).toEqual("search");
+      expect(location?.search).toEqual("?q=BC");
+    });
+    it("should render 'No Results Found' when the query has no results", async () => {
```

Comment:
nit: not a single space between `it` unlike other `it`s in the same block. In fact, I don't really care if there's space or not, but they all should either have a space in between or not at all for consistency.

Also, maybe this isn't something we should keep in mind since the linter/formatter should take care of this e.g. https://github.com/dangreenisrael/eslint-plugin-jest-formatting